core/src/main/java/hudson/tasks/Maven.java

```diff
@@ -137,6 +141,15 @@
      */
     private GlobalSettingsProvider globalSettings;
 
+    /**
+     * Skip injecting build variables as properties into maven process.
+     *
+     * Defaults to false to mimic the legacy behavior.
+     *
+     * @since TODO
+     */
+    private boolean doNotInjectBuildVariables = false;
```

Comment:
I used the most descriptive name I can think off, that is the only think that matters for a field name IMO.